plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -313,73 +316,125 @@ public void checkUnusedEntry(final Entry entry) {
 				&& entry.getIncomingTransitions().isEmpty()) {
 			org.yakindu.sct.model.sgraph.State state = (org.yakindu.sct.model.sgraph.State) entry.getParentRegion()
 					.getComposite();
-			if (!entry.isDefault()) {
-				boolean hasIncomingTransition = false;
-				Iterator<Transition> transitionIt = state.getIncomingTransitions().iterator();
-				while (transitionIt.hasNext() && !hasIncomingTransition) {
-					Iterator<ReactionProperty> propertyIt = transitionIt.next().getProperties().iterator();
-					while (propertyIt.hasNext() && !hasIncomingTransition) {
-						ReactionProperty property = propertyIt.next();
-						if (property instanceof EntryPointSpec) {
-							hasIncomingTransition = entry.getName().equals(((EntryPointSpec) property).getEntrypoint());
-						}
+			if (entry.isDefault()) {
+				return;
+			}
+			boolean hasIncomingTransition = false;
+			Iterator<Transition> transitionIt = state.getIncomingTransitions().iterator();
+			while (transitionIt.hasNext() && !hasIncomingTransition) {
+				Iterator<ReactionProperty> propertyIt = transitionIt.next().getProperties().iterator();
+				while (propertyIt.hasNext() && !hasIncomingTransition) {
+					ReactionProperty property = propertyIt.next();
+					if (property instanceof EntryPointSpec) {
+						hasIncomingTransition = entry.getName().equals(((EntryPointSpec) property).getEntrypoint());
 					}
 				}
-				if (!hasIncomingTransition) {
-					warning(ENTRY_UNUSED, entry, null, -1);
+			}
+			if (!hasIncomingTransition) {
+				warning(ENTRY_UNUSED, entry, null, -1);
+			}
+		}
+
+	}
+
+	@Check(CheckType.FAST)
+	public void checkNotRaisedOutEvent(EventDefinition event) {
+		if (event.eIsProxy() || event.getDirection() != Direction.OUT
+				|| !(event.eContainer() instanceof InterfaceScope)) {
+			return;
+		}
+		// TODO: unit test schreiben
+
+		Statechart statechart = utils.getStatechart(event);
+		List<EventRaisingExpression> eventRaisingExpressions = EcoreUtil2.getAllContentsOfType(statechart,
+				EventRaisingExpression.class);
+
+		List<ReactionTrigger> triggers = EcoreUtil2.getAllContentsOfType(statechart, ReactionTrigger.class);
+
+		boolean isRaisedInStatechart = eventRaisingExpressions.stream()
+				.anyMatch(exp -> EcoreUtil.equals(event, getRaisedEvent(exp)));
+		boolean isUsedAsReactionTrigger = false;
+		for (ReactionTrigger trigger : triggers) {
+			if (trigger.eContainer() instanceof Transition) {
+				Transition transition = (Transition) trigger.eContainer();
+				// event names are unique
+				if (transition.getSpecification().trim().equals(event.getName())) {
+					isUsedAsReactionTrigger = true;
 				}
 			}
 		}
+
+		boolean isUsedInsideReactionGuards = false;
+		if ((isUsedAsReactionTrigger || isUsedInsideReactionGuards) && !isRaisedInStatechart) {
+			warning(String.format(OUT_EVENT_NEVER_RAISED, event.getName()), null, -1);
+		}
+	}
+
+	private EventDefinition getRaisedEvent(EventRaisingExpression exp) {
```

Comment:
protected, also this belongs into an extension, for example the already injected STextExtensions.